integrations/opennms-snmp-hardware-inventory-provisioning-adapter/src/main/java/org/opennms/netmgt/provision/SnmpHardwareInventoryProvisioningAdapter.java

```diff
@@ -221,15 +222,25 @@ private void initializeVendorAttributes() {
             LOG.debug("Loading attribute type {}", type);
             m_vendorAttributes.put(type.getSnmpObjId(), type);
         }
+        //changed implementation to resolve NMS-14085
+        //New implementation compares existing values of hwentityattributetype and values coming
+        // from snmp-hardware-inventory-adapter-configuration.xml , if value is not in table and its new
+        // entry in xml then insert record in table. If value is already in table then update it
+        // with current information (oid, alias, class) coming from xml file.
         for (HwExtension ext : m_hwInventoryAdapterConfigDao.getConfiguration().getExtensions()) {
             for (MibObj obj : ext.getMibObjects()) {
                 HwEntityAttributeType type = m_vendorAttributes.get(obj.getOid());
                 if (type == null) {
                     type = new HwEntityAttributeType(obj.getOid().toString(), obj.getAlias(), obj.getType());
                     LOG.info("Creating attribute type {}", type);
-                    m_hwEntityAttributeTypeDao.save(type);
-                    m_vendorAttributes.put(type.getSnmpObjId(), type);
+                } else {
+                    type.setOid(obj.getOid().toString());
```

Comment:
Do we have a test for this ?

Reply:
Test case added to SnmpHardwareInventoryProvisioningAdapterIT.java